prow/plugins/commands/commands_test.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package commands
+
+import (
+	"errors"
+	"regexp"
+	"testing"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+func TestGenericCommentHandler(t *testing.T) {
+	re := regexp.MustCompile(`foo`)
+
+	tcs := []struct {
+		name         string
+		issueType    IssueType
+		event        github.GenericCommentEvent
+		shouldHandle bool
+		shouldError  bool
+	}{
+		{
+			name:      "issue happy case",
```

Comment:
Please include the expected result of the scenario, not just the scenario, so "issue handler does not respond to PRs" rather than "issue handler on PR"